src/java/org/apache/cassandra/index/sai/plan/TopKProcessor.java

```diff
@@ -258,12 +182,11 @@ private <U extends Unfiltered, R extends BaseRowIterator<U>, P extends BaseParti
 
                 }
             }
+            // reorder rows in partition/clustering order
+            for (var triple : topK.getUnsortedShared())
+                addUnfiltered(unfilteredByPartition, triple.getLeft(), triple.getMiddle());
```

Comment:
As mentioned in https://github.com/datastax/cassandra/pull/1488, [here](https://github.com/datastax/cassandra/blob/efcc375a0f00cb40a80fd69670efc86c3ca82387/src/java/org/apache/cassandra/index/sai/plan/TopKProcessor.java#L177-L183), I think we don't need to reorder rows for the coordinator sorting. Not even sure if we even need any of that ordering on the replicas given that we only support CL=ONE, so there is no reconciliation. But we can leave it for CNDB-12308, which will need some rebase work if it comes after this.